nomos-services/data-availability/network/src/storage/adapters/mock.rs

```diff
@@ -11,17 +11,20 @@ pub struct MockStorageService;
 
 impl ServiceData for MockStorageService {
     type Settings = ();
-
     type State = NoState<()>;
-
     type StateOperator = ();
-
     type Message = ();
 }
 
+#[derive(Default)]
+struct StorageState {
+    assignations: HashMap<BlockNumber, Assignations<PeerId, SubnetworkId>>,
+    addressbooks: HashMap<BlockNumber, HashMap<PeerId, Multiaddr>>,
+}
+
 #[derive(Default)]
 pub struct MockStorage {
-    storage: Mutex<HashMap<BlockNumber, Assignations<PeerId, SubnetworkId>>>,
+    state: Mutex<StorageState>,
 }
 
 impl MembershipStorageAdapter<PeerId, SubnetworkId> for MockStorage {
```

Comment:
This is fine. I think there is something missing which is a call for prunning old states that wont be required anymore. I can't say now when and how it should be triggered, but we would have to do it eventually.